File: LeetCode/Easy/0599-minimum-index-sum-of-two-lists/0599-minimum-index-sum-of-two-lists.cpp
class Solution {
public:
    vector<string> findRestaurant(vector<string>& list1, vector<string>& list2) {
        unordered_map<string, int> exist;
        
        vector<string> ret;
        int minVal = INT_MAX;
        int j;
        for (int i = 0; i < list1.size(); ++i) exist[list1[i]] = i;
        
        for (int i = 0; i < list2.size(); ++i){
            if (exist.find(list2[i]) != exist.end()){
                j = exist[list2[i]];
                if (j + i < minVal){
                    minVal = j + i;
                    ret.clear();
                    ret.push_back(list2[i]);
                } else if (j + i == minVal) ret.push_back(list2[i]);
            }
        }
        
        return ret;
    }
};

File: LeetCode/Easy/0599-minimum-index-sum-of-two-lists/0599-minimum-index-sum-of-two-lists.py
class Solution:
    def findRestaurant(self, list1: List[str], list2: List[str]) -> List[str]:
        exist = {}

        temp = [[] for i in range(len(list1) + len(list2))]

        for i in range(len(list1)):
            exist[list1[i]] = i
        
        for i in range(len(list2)):
            if list2[i] in exist:
                temp[exist[list2[i]] + i].append(list2[i])

        for v in temp:
            if v:
                return v
        